pkg/controller/factory/controller_context.go

```diff
@@ -18,21 +18,24 @@ import (
 // the sync to be triggered.
 type syncContext struct {
 	eventRecorder events.Recorder
-	queue         workqueue.RateLimitingInterface
+	rl            workqueue.RateLimiter
+	queue         addRateLimitedAfterWorkqueue
 	queueKey      string
 }
 
 var _ SyncContext = syncContext{}
 
 // NewSyncContext gives new sync context.
 func NewSyncContext(name string, recorder events.Recorder) SyncContext {
+	rl := workqueue.DefaultControllerRateLimiter()
 	return syncContext{
-		queue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), name),
+		rl:            rl,
+		queue:         addRateLimitedAfterWorkqueue{workqueue.NewNamedRateLimitingQueue(rl, name), rl},
 		eventRecorder: recorder.WithComponentSuffix(strings.ToLower(name)),
 	}
 }
 
-func (c syncContext) Queue() workqueue.RateLimitingInterface {
+func (c syncContext) Queue() RateLimitedAddAfterWorkqueue {
```

Comment:
this will still make it default to your queue right? I mean all those 100+ controllers we run will switch to this after we bump :-)

Reply:
it won't. The method is new.